From 2ff05e5ae2ea59200f6b3d0a18ed00fe35e690f5 Mon Sep 17 00:00:00 2001
From: Ranjan Dutta <ranjan.dutta@intel.com>
Date: Tue, 10 Aug 2021 09:42:26 +0800
Subject: [PATCH 04/11] Revert "drm/amdgpu: Avoid printing of stack contents on
 firmware load error"

This reverts commit ccc7a1bb322e4169b738aa3fc8b5dfcfefdacb37.
---
 drivers/gpu/drm/amd/amdgpu/psp_v12_0.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/psp_v12_0.c b/drivers/gpu/drm/amd/amdgpu/psp_v12_0.c
index b0ee77ee80b9..c4828bd3264b 100644
--- a/drivers/gpu/drm/amd/amdgpu/psp_v12_0.c
+++ b/drivers/gpu/drm/amd/amdgpu/psp_v12_0.c
@@ -67,7 +67,7 @@ static int psp_v12_0_init_microcode(struct psp_context *psp)
 
 	err = psp_init_asd_microcode(psp, chip_name);
 	if (err)
-		return err;
+		goto out;
 
 	snprintf(fw_name, sizeof(fw_name), "amdgpu/%s_ta.bin", chip_name);
 	err = request_firmware(&adev->psp.ta_fw, fw_name, adev->dev);
@@ -80,7 +80,7 @@ static int psp_v12_0_init_microcode(struct psp_context *psp)
 	} else {
 		err = amdgpu_ucode_validate(adev->psp.ta_fw);
 		if (err)
-			goto out;
+			goto out2;
 
 		ta_hdr = (const struct ta_firmware_header_v1_0 *)
 				 adev->psp.ta_fw->data;
@@ -105,9 +105,10 @@ static int psp_v12_0_init_microcode(struct psp_context *psp)
 
 	return 0;
 
-out:
+out2:
 	release_firmware(adev->psp.ta_fw);
 	adev->psp.ta_fw = NULL;
+out:
 	if (err) {
 		dev_err(adev->dev,
 			"psp v12.0: Failed to load firmware \"%s\"\n",
-- 
2.27.0

